### PR TITLE
[c10/**] Fix typos

### PR DESCRIPTION
**Pull request by SamuelMarks**
All pretty minor. I avoided renaming `class DestructableMock` to `class DestructibleMock` and similar such symbol renames (in this PR).

### CODE REVIEW

File: c10/util/complex.h [hunk 1]
```diff
@@ -61,7 +61,7 @@ namespace c10 {
 //     Since we only support float and double, on will use `complex& operator=(T x)`
 // - Copy assignment operator and converting assignment operator
 //   - There is no specialization of converting assignment operators, which type is
-//     convertible is soly depend on whether the scalar type is convertable
+//     convertible is solely dependant on whether the scalar type is convertible
```
Comment (mruberry) [inline c10/util/complex.h:64]: "dependent" -> "dependent"

File: c10/core/Allocator.h [hunk 2]
```diff
@@ -74,7 +74,7 @@ class C10_API DataPtr {
    * However, in some cases, you know /exactly/ what the deleter
    * is, and you have a new deleter that manually wraps
    * the old one.  In this case, you can safely swap the deleter
-   * after asserting that the deleters line up.
+   * after asserting that the deleters line is up.
```
Comment (mruberry) [inline c10/core/Allocator.h:77]: I think "line up" in the original is intended

**Review by mruberry — CHANGES_REQUESTED** [review]
Hey @SamuelMarks, thanks for submitting this PR. Spelling corrections to user-facing code, like that in the TORCH_CHECK macros, are welcome. Spelling improvements in developer-facing code are probably too inconsequential to be merged (see [here](https://github.com/pytorch/pytorch/wiki/Developer-FAQ#my-pr-was-rejected-for-being-too-small--what-gives)). They can be included in this PR since it's fixing two user-facing spelling errors, however.

This PR does have two mistakes, however. These changes can just be reverted for simplicity.

File: c10/core/Allocator.h (hunk 2 above)
Comment (SamuelMarks) [reply c10/core/Allocator.h]: Ahh yep, I see, fixed for clarity

File: c10/core/Allocator.h [hunk 3]
```diff
@@ -74,7 +74,7 @@ class C10_API DataPtr {
    * However, in some cases, you know /exactly/ what the deleter
    * is, and you have a new deleter that manually wraps
    * the old one.  In this case, you can safely swap the deleter
-   * after asserting that the deleters line up.
+   * after asserting that the deleters line-up.
```
Comment (mruberry) [inline c10/core/Allocator.h:77]: Hyphenating this is incorrect because then it would be an adjective. This "line up" is intended to be a verb.

Comment (SamuelMarks) [reply c10/core/Allocator.h:77]: Hmm, looks like it's also valid as one word: https://www.merriam-webster.com/dictionary/lineup
> an alignment (as in entertainment or politics) of persons or things having a common purpose, distinction, or bond

**Review by mruberry — APPROVED** [review]
Thanks @SamuelMarks.

**Pull request closed without merging**